src/diffusers/loaders/lora_base.py

```diff
@@ -661,9 +670,18 @@ def unfuse_lora(self, components: List[str] = [], **kwargs):
                 if issubclass(model.__class__, (ModelMixin, PreTrainedModel)):
                     for module in model.modules():
                         if isinstance(module, BaseTunerLayer):
+                            for adapter in set(module.merged_adapters):
+                                if adapter and adapter in self._merged_adapters:
+                                    self._merged_adapters.remove(adapter)
```

Comment:
Same idea as in `fuse_lora` to avoid mutating a class attribute. It's probably not strictly necessary because `unfuse_lora` is never called before `fuse_lora` but IMO it's better to be rigorous here.

```suggestion
                                    self._merged_adapters = self._merged_adapters - {adapter}
```